src/compiler/comments.ts

```diff
@@ -269,6 +287,18 @@ namespace ts {
             currentText = currentSourceFile.text;
             currentLineMap = getLineStarts(currentSourceFile);
             detachedCommentsInfo = undefined;
+            disabled = false;
+        }
+
+        function disableCommentsAndEmit(node: Node, emitCallback: (node: Node) => void): void {
+            if (disabled) {
+                emitCallback(node);
+            }
+            else {
+                disabled = true;
+                emitCallback(node);
+                disabled = false;

```

Comment:
maybe it should save previous value of `disabled` and restore it later instead of unconditionally set it to `false`?